apis/iam/v1beta1/serviceaccount_types.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// ServiceAccountParameters defines parameters for a desired IAM ServiceAccount
+// https://cloud.google.com/iam/docs/reference/rest/v1/projects.serviceAccounts
+type ServiceAccountParameters struct {
+	// Optional. A user-specified name for the service account.
+	// Must be less than or equal to 100 UTF-8 bytes.
+	// +optional
+	DisplayName *string `json:"displayName,omitempty"`
+
+	// Optional. A user-specified opaque description of the service account.
+	// Must be less than or equal to 256 UTF-8 bytes.
+	// +optional
+	Description *string `json:"description,omitempty"`
+}
+
+// ServiceAccountObservation is used to show the observed state of the
+// ServiceAccount resource on GCP.
+type ServiceAccountObservation struct {
+	// The resource name of the service account in the following format:
+	// projects/{PROJECT_ID}/serviceAccounts/{ACCOUNT}.
+	// part of https://godoc.org/google.golang.org/genproto/googleapis/iam/admin/v1#ServiceAccount
+	// not to be confused with CreateServiceAccountRequest.Name aka ServiceAccountParameters.ProjectName
+	// +immutable
+	Name string `json:"name,omitempty"`
+
+	// The id of the project that owns the service account.
+	// Note: This field is used in responses only. Any value specified here
```

Comment:
Fair enough, I suppose existing in the `Status` field makes that clear on its own once you are familiar with our idioms. Although, for new users who are bottom-up learners, I worry that our type-level documentation sometimes relies too heavily on having already grasped the top-down concepts, especially when those concepts are most clearly conveyed in developer documentation (like Status' immutabilty). I might put a comment on Status to this effect, redundant as it may be.